fahrschein/src/main/java/org/zalando/fahrschein/NakadiClient.java

```diff
@@ -139,8 +134,9 @@ public Subscription subscribe(String applicationName, Set<String> eventNames, St
      * @return
      * @throws IOException
      */
-    public Subscription subscribe(String applicationName, Set<String> eventNames, String consumerGroup, SubscriptionRequest.Position readFrom) throws IOException {
-        final SubscriptionRequest subscription = new SubscriptionRequest(applicationName, eventNames, consumerGroup, readFrom);
+    public Subscription subscribe(String applicationName, Set<String> eventNames, String consumerGroup, SubscriptionRequest.Position readFrom, Set<Cursor> initialCursors) throws IOException {
```

Comment:
I'm wondering if a `Map<String, List<Cursor>>` with event name as key makes more sense in the api, to avoid repeating the event name for each partition. The initial cursors are also only valid if `Position` is indeed `cursors`, so maybe there should be some validation in place.